scripts/make_confusion_matrix.py

```diff
@@ -0,0 +1,89 @@
+import argparse
+import json
+from pathlib import Path
+
+import numpy as np
+import sklearn
+import sklearn.metrics
+
+
+allowed_metrics = ["acc", "acc_norm"]
+
+
+def parse_args():
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "--file",
+        required=True,
+        metavar="DIR|DIR/file.json",
+        help="Path of json file containing sample predictions.",
+    )
+    parser.add_argument(
+        "--metric",
+        default="acc",
+        choices=allowed_metrics,
+        help="Method of determining prediction.",
+    )
+    return parser.parse_args()
+
+
+def confusion_matrix(file: str, metric: str):
+    file = Path(file)
+    assert file.exists(), f"File does not exist! ({file})"
+
+    sample_dump = []
+    with open(file, "r") as f:
+        for line in f:
+            sample = json.loads(line)
+            sample_dump.append(sample)
+
+    choices = [d["arg_1"].strip() for d in sample_dump[0]["arguments"].values()]
+    completion_len = np.array([float(len(i)) for i in choices])
+    print(f"Classes: {choices}")
+
+    targets, predictions = [], []
+    for sample in sample_dump:
+        target = sample["target"]
+        if isinstance(target, int):
+            target_label = choices[target]
+            targets.append(target_label)
+        elif isinstance(target, str):
+            assert target in choices
+            targets.append(target)
+        elif isinstance(target, list):
+            raise NotImplementedError(
+                "No support yet for multi-label confusion matrix!"
+            )
+
+        results = sample["resps"]
+        lls = [float(r[0][0]) for r in results]
+        pred = np.argmax(lls)
+        pred_norm = np.argmax(lls / completion_len)
+        if metric == "acc":
+            pred_label = choices[pred]
+        elif metric == "acc_norm":
+            pred_label = choices[pred_norm]
+        predictions.append(pred_label)
```

Comment:
Prediction method (`acc` or `acc_norm`) copied based on [this](https://github.com/EleutherAI/lm-evaluation-harness/blob/6f7b4a0588bc2217b4a3c25aa1f2ffdb02daa3b9/lm_eval/api/task.py#L1375-L1376). A drawback of this is that if the original method changes (e.g. normalization constant changes from number of characters to number of tokens), this will not be reflected here.